api/python/quilt3/data_transfer.py

```diff
@@ -741,15 +741,25 @@ def list_object_versions(bucket, prefix, recursive=True):
     s3_client = S3ClientProvider().find_correct_client(S3Api.LIST_OBJECT_VERSIONS, bucket, list_obj_params)
     paginator = s3_client.get_paginator('list_object_versions')
 
-    for response in paginator.paginate(**list_obj_params):
-        versions += response.get('Versions', [])
-        delete_markers += response.get('DeleteMarkers', [])
-        prefixes += response.get('CommonPrefixes', [])
+    try:
```

Comment:
I'm not sure that's the right place to have this logic, might be a better idea to have it `Package.set_dir()`

Reply:
Fair point. Moving.